lib/common/variadic/uint32OrHash.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2021 ChainSafe Systems (ON)
```

Comment:
Git maniac, forgive me: can you rebase interactively your branch and have 1 commit to move the files uint64OrHash -> uint32OrHash, and then 1 commit with the rest of your changes? That way we can rebase and merge the PR (not squash and merge) so file movements are tracked and they'll be less deltas/better blame/better rebasing.

Let me know I can do it on your branch if you want.

Reply:
honestly I don't usually rebase... feel free to do it for me haha, thank you!

Reply:
Done, you should run `git checkout noot/fix-block-request && git reset --hard origin/noot/fix-block-request` to get the rewritten history.

For future changes, just stage your changes and run `git commit --amend --no-edit` to bundle it in the second commit, and force push to override the 2nd commit.

Reply:
awesome thank you!!!